test/lib/dockerfile/index.spec.ts

```diff
@@ -0,0 +1,36 @@
+import { DockerfileParser } from "dockerfile-ast";
+import { EOL } from "os";
+import { getDockerfileBaseImageName } from "../../../lib/dockerfile";
+
+describe("base image parsing", () => {
+  it("does not detect injected base images", () => {
+    const dockerfiles = [
+      "FROM ${A}:${B}",
+      "ARG A" + EOL + "ARG B" + EOL + "FROM ${A}:${B}",
+      "ARG A" + EOL + "FROM ${A}:tag",
+      "ARG B" + EOL + "FROM alpine:${B}",
+      "ARG A" + EOL + "ARG B" + EOL + "FROM ${A}:${B} AS image",
+    ];
+    for (const dockerfile of dockerfiles) {
+      expect(
+        getDockerfileBaseImageName(DockerfileParser.parse(dockerfile)),
+      ).toBeUndefined();
+    }
+  });
+
+  it("detects various base images", () => {
+    const dockerfiles = [
+      "FROM image",
+      "FROM image AS foo",
+      "FROM image:tag",
+      "FROM image:tag AS foo",
+      "FROM image@sha256:abcd",
+      "FROM image@sha256:abcd AS foo",
+    ];
+    for (const dockerfile of dockerfiles) {
+      expect(
+        getDockerfileBaseImageName(DockerfileParser.parse(dockerfile)),
+      ).toBeDefined();
+    }
```

Comment:
Did we also think about using [it.each()](https://jestjs.io/docs/en/api#testeachtablename-fn-timeout) here, so the test report shows a separate test name for each individual dockerfile? I believe this helps with parallelisation and the "running failures first" stuff Jest does to speed up the feedback cycle too. Not sure how important those considerations are here though, esp since the for..of loop is arguably easier to read than the it.each() syntax 🤔